components/PaginationSection.tsx

```diff
@@ -0,0 +1,91 @@
+import { ChevronLeftIcon, ChevronRightIcon } from '@chakra-ui/icons'
+import { HStack, IconButton, Stack, StackProps, Text } from '@chakra-ui/react'
+import Select from 'react-select'
+import { usePaginationContext } from 'hooks'
+
+const pageRangeSize = 10
+const getPageRange = (currPage: number, numPages: number): [number, number] => {
+  if (currPage + pageRangeSize - 1 >= numPages) {
+    return [Math.max(1, numPages - pageRangeSize + 1), numPages]
+  }
+
+  // Shift start page to current page when current page is over the middle of the range
+  const halfPageRangeSize = pageRangeSize / 2
+  const startPage =
+    currPage % halfPageRangeSize === 1
+      ? currPage
+      : currPage % halfPageRangeSize === 0
+      ? currPage - halfPageRangeSize + 1
+      : Math.floor(currPage / halfPageRangeSize) * halfPageRangeSize + 1
+  const endPage = Math.min(startPage + 2 * halfPageRangeSize - 1, numPages)
+
+  return [startPage, endPage]
+}
+
+export const PaginationSection = (props: StackProps): JSX.Element => {
+  const { page, numPages, setPage, pageSizeOptions, setPageSize } =
+    usePaginationContext()
```

Comment:
This component and its associated hooks give us more control over functionality and styling than using [React Paginated List](https://github.com/rigelglen/react-paginated-list).